src/System.Xml.XmlSerializer/src/System/Xml/Serialization/ReflectionXmlSerializationWriter.cs

```diff
@@ -0,0 +1,1112 @@
+                                WriteAttribute(attribute.Name, ns, sb.ToString());
+                            }
+                        }
+                    }
+                }
+            }
+            else
+            {
+                WriteAttribute(memberValue, attribute, parent);
+            }
+        }
+
+        bool CanOptimizeWriteListSequence(TypeDesc listElementTypeDesc) {
+            // check to see if we can write values of the attribute sequentially
+            // currently we have only one data type (XmlQualifiedName) that we can not write "inline", 
+            // because we need to output xmlns:qx="..." for each of the qnames
+            return (listElementTypeDesc != null && listElementTypeDesc != QnameTypeDesc);
+        }
+
+        private void WriteAttribute(object memberValue, AttributeAccessor attribute, object parent)
+        {
+            if (attribute.Mapping is SpecialMapping)
+            {
+                // TODO: this block is never hit by our tests.
+                SpecialMapping special = (SpecialMapping)attribute.Mapping;
+                if (special.TypeDesc.Kind == TypeKind.Attribute || special.TypeDesc.CanBeAttributeValue)
+                {
+                    WriteXmlAttribute((XmlNode)memberValue, parent);
+                }
+                else
+                    throw new InvalidOperationException(SR.XmlInternalError);
+            }
+            else
+            {
+                string ns = attribute.Form == XmlSchemaForm.Qualified ? attribute.Namespace : "";
+                WritePrimitive("WriteAttribute", attribute.Name, ns, attribute.Default, memberValue, attribute.Mapping, false, false, false);
+            }
+        }
+
+        private int FindXmlnsIndex(MemberMapping[] members)
+        {
+            for (int i = 0; i < members.Length; i++)
+            {
+                if (members[i].Xmlns == null)
+                    continue;
+                return i;
+            }
+            return -1;
+        }
+
+        bool WriteDerivedTypes(StructMapping mapping, string n, string ns, object o, bool isNullable)
+        {
+            Type t = o.GetType();
+            for (StructMapping derived = mapping.DerivedMappings; derived != null; derived = derived.NextDerivedMapping)
+            {
+                if (t == derived.TypeDesc.Type)
+                {
+                    WriteStructMethod(derived, n, ns, o, isNullable, needType: true);
+                    return true;
+                }
+
+                if (WriteDerivedTypes(derived, n, ns, o, isNullable))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private void WritePrimitive(string method, string name, string ns, object defaultValue, object o, TypeMapping mapping, bool writeXsiType, bool isElement, bool isNullable)
```

Comment:
>string method [](start = 36, length = 13)

method should be of type WritePrimitiveMethod, the enum at the bottom of this file.